code/controllers/subsystem/aura.dm

```diff
@@ -0,0 +1,98 @@
+/// Tracks aura emitters and tells them to pulse. Also used for creating new ones.
+SUBSYSTEM_DEF(aura)
+	name = "Auras"
+	wait = 2 SECONDS
+	flags = SS_NO_INIT
+	///Stores currently active aura_bearer datums
+	var/list/active_auras = list() //We can't use a normal subsystem processing list because as soon as an aura_bearer leaves the list it needs to die
+
+/datum/controller/subsystem/aura/fire(resumed)
+	for(var/datum/aura_bearer/bearer AS in active_auras)
+		bearer.process()
+
+///Use this to start a new emitter with the specified stats. Returns the emitter in question, use stop_emitting() to end early.
+/datum/controller/subsystem/aura/proc/add_emitter(atom/center, type, range, strength, duration)
```

Comment:
you should be able to emit multiple auras at once without doing the range finding twice